gui/src/hooks/useRefreshSettings.ts

```diff
@@ -0,0 +1,5 @@
+import { useEventListener } from "./useEventListener";
+
+export function useRefreshSettings(callback: () => unknown) {
+  useEventListener("settings-changed", callback);
```

Comment:
we could just type type `useEventListener` with the union of all supported events, and remove all these `useRefresh***` indirection?

Reply:
I didn't get what you mean

Reply:
that instead of `useRefreshSettings`, `useRefreshTransaction`, and all these others, we can just use `useEventListener` directly

and we type it as:

```typescript
type Event = "settings-changed" | "transactions-updated" | ...;
useEventListener(event: Event, ...)`;
```

Reply:
I guess these listeners will always be the same 👍

Reply:
done